### PR TITLE
Linux CPU unit test workflow using GitHub Actions

### PR DESCRIPTION
**Pull request by robandpdx**
## Description

Migrate unit tests from CircleCI to GitHub Actions

## Motivation and Context

Why is this change required? What problem does it solve?
The direction is to move unit tests to GitHub Actions. 

- [ ] I have raised an issue to propose this change ([required](https://github.com/pytorch/rl/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

No change to functionality.

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x ] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .github/workflows/test-linux-cpu.yml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        env_dir="${work_dir}/env"
+        lib_dir="${env_dir}/lib"
+
+        # 1. Install conda at ./conda
+        if [ ! -d "${conda_dir}" ]; then
+            printf "* Installing conda\n"
+            wget -O miniconda.sh "http://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh"
+            bash ./miniconda.sh -b -f -p "${conda_dir}"
+        fi
+        eval "$(${conda_dir}/bin/conda shell.bash hook)"
```
Comment (malfet) [inline .github/workflows/test-linux-cpu.yml:46]: Those containers should already have conda installed

Comment (robandpdx) [reply .github/workflows/test-linux-cpu.yml]: @malfet Thanks for the help today. The workflow is green now. Moving forward as planned.

**Comment by DanilBaibak** [comment]
Hi @vmoens! @robandpdx works on  migration the "Unit-tests on Linux GPU" job to GitHub Actions.

**Comment by DanilBaibak** [comment]
Hi @vmoens! @robandpdx works on migration the "Unit-tests on Linux GPU" job to GitHub Actions. There is a problem with the smoke test `smoke_test_deps.py::test_dm_control`: [ImportError: Cannot initialize a headless EGL](https://github.com/pytorch/rl/actions/runs/3955105929/jobs/6773094082#step:10:3116).  Did you face such issue before?

**Comment by vmoens** [comment]
> Hi @vmoens! @robandpdx works on migration the "Unit-tests on Linux GPU" job to GitHub Actions. There is a problem with the smoke test `smoke_test_deps.py::test_dm_control`: [ImportError: Cannot initialize a headless EGL](https://github.com/pytorch/rl/actions/runs/3955105929/jobs/6773094082#step:10:3116). Did you face such issue before?

It's a common issue with dm_control and related. We provide some solutions here:
https://pytorch.org/rl/reference/generated/knowledge_base/MUJOCO_INSTALLATION.html#installing-mujoco

**Pull request merged**